File: jscomp/test/test_sprintf.js
'use strict';

var Curry = require("../../lib/js/curry.js");
var Printf = require("../../lib/js/printf.js");

console.error(Curry._2(Printf.sprintf(/* Format */{
              _0: {
                TAG: /* Int */4,
                _0: /* Int_d */0,
                _1: /* No_padding */0,
                _2: /* No_precision */0,
                _3: {
                  TAG: /* String */2,
                  _0: /* No_padding */0,
                  _1: /* End_of_format */0
                }
              },
              _1: "%d%s"
            }), 32, "ss"));

/*  Not a pure module */

File: jscomp/test/test_sprintf.ml
(* (Printf.sprintf *)
(* (/* Format */[0, *)
(* /* Int */[4, *)
(* /* Int_d */0, *)
(* /* No_padding */0, *)
(* /* No_precision */0, *)
(* /* String */[2,/* No_padding */0,/* End_of_format */0]], *)
(* "%d%s"]) *)
(* (32,"ss")) *)

(* working version *)
(* (Printf.sprintf *)
(* (/* Format */[0, *)
(* /* Int */[4, *)
(* /* Int_d */0, *)
(* /* No_padding */0, *)
(* /* No_precision */0, *)
(* /* String */[2,/* No_padding */0,/* End_of_format */0]], *)
(*       "%d%s"]) *)
(*    (32)("ss")) *)

;;
prerr_endline @@ Printf.sprintf "%d%s" 32 "ss"
